### PR TITLE
ADD keyboard Projectd/65/Projectd_65_ ansi

### PR DESCRIPTION
**Pull request by gskygithub**
## Description

ADD project_65_ansi keyboard 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/projectd/65/projectd_65_ansi/info.json [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        ]
+    },
+    "url": "",
+    "usb": {
+        "device_version": "0.0.4",
+        "pid": "0x5319",
+        "vid": "0x3233"
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/projectd/65/projectd_65_ansi/info.json:160]: matching community layout identified
```suggestion
    },
    "layout_aliases": {
        "LAYOUT": "LAYOUT_65_ansi_blocker"
    },
    "community_layouts": ["65_ansi_blocker"],
    "layouts": {
        "LAYOUT_65_ansi_blocker": {
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
is `darkproject/kd83a_bfg_edition` directory intended to included in this PR?

**Comment by gksygithub** [comment]
> is `darkproject/kd83a_bfg_edition` directory intended to included in this PR?

No, It's mistakenly included.

**Comment by dunk2k** [comment]
> > is `darkproject/kd83a_bfg_edition` directory intended to included in this PR?
> 
> No, It's mistakenly included.

Thanks for response. Could you remove named directory from source branch and push commit? so that named directory is no longer included in this PR.

File: keyboards/projectd/65/projectd_65_ansi/info.json [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            { "flags": 4, "matrix": [0, 3], "x": 130, "y": 40 },
+            { "flags": 4, "matrix": [1, 5], "x": 140, "y": 40 },
+            { "flags": 4, "matrix": [2, 5], "x": 150, "y": 40 }
+        ]
+    },
+    "url": "",
+    "usb": {
+        "device_version": "0.0.4",
+        "pid": "0x5319",
+        "vid": "0x3233"
```
Comment (drashna) [inline keyboards/projectd/65/projectd_65_ansi/info.json:157]: This vendor ID belongs to DuckyChannel, and I suspect that this board isn't theirs.
https://www.the-sz.com/products/usbid/index.php?v=0x3233&p=&n=

Please change the Vendor ID.  Generally, 0x4000 and higher isn't assigned, and the further past this, the better.

File: keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+enum custom_layers {
```
Comment (drashna) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c:21]: Might just be better to use numbers and not use the enum here, at all.

File: keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,            KC_UP,    KC_END,
+            KC_LCTL,  KC_LGUI,  KC_LALT,                      KC_SPC,                                 KC_RALT,  MO(Layer_1),        KC_LEFT,  KC_DOWN,  KC_RGHT),
+[Layer_1] = LAYOUT(
+            KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_DEL,   KC_NO,
+            QK_BOOT,  KC_NO,    KC_NO,    KC_NO,    RGB_TOG,  RGB_MOD,  RGB_RMOD, KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_LSFT,  KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_RSFT,            RGB_VAI,  KC_NO,
+            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              RGB_SPD,  RGB_VAD,  RGB_SPI),
+[Layer_2] = LAYOUT(
```
Comment (drashna) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c:52]: Empty layers should/need to be removed.

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+            KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,            KC_UP,    KC_END,
+            KC_LCTL,  KC_LGUI,  KC_LALT,                      KC_SPC,                                 KC_RALT,  MO(Layer_1),        KC_LEFT,  KC_DOWN,  KC_RGHT),
+[Layer_1] = LAYOUT(
+            KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_DEL,   KC_NO,
+            QK_BOOT,  KC_NO,    KC_NO,    KC_NO,    RGB_TOG,  RGB_MOD,  RGB_RMOD, KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_LSFT,  KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_RSFT,            RGB_VAI,  KC_NO,
+            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              RGB_SPD,  RGB_VAD,  RGB_SPI),
+[Layer_2] = LAYOUT(
```
Comment (drashna) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c:103]: Especially here.  Any unpopulated layer gets filled automatically with transparencies.  So blank layers are not needed, anymore.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/projectd/65/projectd_65_ansi/info.json (hunk 2 above)
Comment (gskygithub) [reply keyboards/projectd/65/projectd_65_ansi/info.json:157]: yes, this keyboard is from ducky's production. So, we keep this VID.

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c (hunk 5 above)
Comment (gskygithub) [reply keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c:103]: we want to use VIA with 13 layers. So, we write like this...

Comment (dunk2k) [reply keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c]: I'm in agreement with @drashna, layers without purpose is excessive.

If the intention is for `via` keymap to make available 13 layers in VIA, over the default 4, then please add a `config.h` file inside the `via` directory with the following contents:
```
#pragma once

#define DYNAMIC_KEYMAP_LAYER_COUNT 13
```

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              KC_NO,    KC_NO,    KC_NO),
+[12] = LAYOUT(
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              KC_NO,    KC_NO,    KC_NO),
```
Comment (dunk2k) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c:152]:
```suggestion
            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              RGB_SPD,  RGB_VAD,  RGB_SPI)
```

File: keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              KC_NO,    KC_NO,    KC_NO),
+[12] = LAYOUT(
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,              KC_NO,    KC_NO,
+            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              KC_NO,    KC_NO,    KC_NO),
+
```
Comment (dunk2k) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c:103]:
```suggestion
            KC_NO,    KC_NO,    KC_NO,                        KC_NO,                                  KC_NO,    KC_NO,              RGB_SPD,  RGB_VAD,  RGB_SPI)
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c (hunk 5 above)
Comment (gskygithub) [reply keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c]: @dunk2k by doing the process you mentioned is displaying KC_TRNS key in VIA but we need KC_NO key....
Is there any suggested way to do this...thanks

Comment (gskygithub) [reply keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c]: @lesshonor We changed this part to KC_TRNS now.

File: keyboards/projectd/65/projectd_65_ansi/config.h [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+#define SPI_MISO_PIN A6
+
+#define DRIVER_1_CS A15
+#define DRIVER_2_CS B15
+#define DRIVER_1_EN C13
+#define DRIVER_2_EN C13
+
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 71
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL)
```
Comment (dunk2k) [inline keyboards/projectd/65/projectd_65_ansi/config.h:43]:
```suggestion
#define RGB_MATRIX_LED_COUNT DRIVER_1_LED_TOTAL
```

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/config.h
Comment (dunk2k) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/via/config.h:1]: This file should be removed as dynamic layer key count has been specified in an `info.json` file in a higher directory.

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — APPROVED** [review]

File: keyboards/projectd/65/projectd_65_ansi/config.h [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+#define SPI_MISO_PIN A6
+
+#define DRIVER_1_CS A15
+#define DRIVER_2_CS B15
+#define DRIVER_1_EN C13
+#define DRIVER_2_EN C13
+
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 71
+#define RGB_MATRIX_LED_COUNT DRIVER_1_LED_TOTAL
```
Comment (waffle87) [inline keyboards/projectd/65/projectd_65_ansi/config.h:43]:
```suggestion
#define RGB_MATRIX_LED_COUNT 71
```

File: keyboards/projectd/65/projectd_65_ansi/info.json [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+    },
+    "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "layer_count": 13
+    },
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
```
Comment (waffle87) [inline keyboards/projectd/65/projectd_65_ansi/info.json:17]:
```suggestion
```

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+            } else {
+                host_consumer_send(0);
+            }
+            return false; /* Skip all further processing of this key */
+
+        default:
+            return true; /* Process all other keycodes normally */
+
+    }
+};
```
Comment (waffle87) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c:69]: Are these necessary when they aren't even used anywhere?

File: keyboards/projectd/65/projectd_65_ansi/projectd_65_ansi.c [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+        palSetPadMode(PAL_PORT(SPI_SCK_PIN), PAL_PAD(SPI_SCK_PIN), PAL_MODE_ALTERNATE(SPI_SCK_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST | PAL_WB32_CURRENT_LEVEL3);
+        palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), PAL_MODE_ALTERNATE(SPI_MOSI_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+        palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), PAL_MODE_ALTERNATE(SPI_MISO_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+    }
+}
+
+#endif
+
+
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
```
Comment (waffle87) [inline keyboards/projectd/65/projectd_65_ansi/projectd_65_ansi.c:136]:
```suggestion
bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
        return false;
    }
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+            KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,       KC_SLSH,  KC_RSFT,            KC_UP,    KC_END,
+            KC_LCTL,  KC_LGUI,  KC_LALT,                      KC_SPC,                                 KC_RALT,      MO(1),              KC_LEFT,  KC_DOWN,  KC_RGHT),
+[1] = LAYOUT(
+            KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,        KC_F10,   KC_F11,   KC_F12,   KC_DEL,   KC_TRNS,
+            KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  RGB_TOG,  RGB_MOD,  RGB_RMOD, KC_TRNS,  KC_TRNS,  KC_TRNS,      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+            KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,      KC_TRNS,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+            KC_LSFT,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,      KC_TRNS,  KC_RSFT,            RGB_VAI,  KC_TRNS,
+            KC_TRNS,  KC_TRNS,  KC_TRNS,                      KC_TRNS,                                LT(2,KC_TRNS),KC_TRNS,            RGB_SPD,  RGB_VAD,  RGB_SPI),
```
Comment (waffle87) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/default/keymap.c:86]:
```suggestion
            KC_TRNS,  KC_TRNS,  KC_TRNS,                      KC_TRNS,                                MO(2), KC_TRNS,            RGB_SPD,  RGB_VAD,  RGB_SPI),
```

File: keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+            KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,       KC_SLSH,  KC_RSFT,            KC_UP,    KC_END,
+            KC_LCTL,  KC_LGUI,  KC_LALT,                      KC_SPC,                                 KC_RALT,      MO(1),              KC_LEFT,  KC_DOWN,  KC_RGHT),
+[1] = LAYOUT(
+            KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,        KC_F10,   KC_F11,   KC_F12,   KC_DEL,   KC_TRNS,
+            KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  RGB_TOG,  RGB_MOD,  RGB_RMOD, KC_TRNS,  KC_TRNS,  KC_TRNS,      KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+            KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,      KC_TRNS,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+            KC_LSFT,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,      KC_TRNS,  KC_RSFT,            RGB_VAI,  KC_TRNS,
+            KC_TRNS,  KC_TRNS,  KC_TRNS,                      KC_TRNS,                                LT(2,KC_TRNS),KC_TRNS,            RGB_SPD,  RGB_VAD,  RGB_SPI),
```
Comment (waffle87) [inline keyboards/projectd/65/projectd_65_ansi/keymaps/via/keymap.c:86]:
```suggestion
            KC_TRNS,  KC_TRNS,  KC_TRNS,                      KC_TRNS,                                MO(2), KC_TRNS,            RGB_SPD,  RGB_VAD,  RGB_SPI),
```

**Review by waffle87 — APPROVED** [review]

File: keyboards/projectd/65/projectd_65_ansi/info.json (hunk 2 above)
Comment (drashna) [reply keyboards/projectd/65/projectd_65_ansi/info.json:156]: Well, there is no indication of that anywhere here, aside from the Vendor ID.

**Review by drashna — APPROVED** [review]

**Comment by gskygithub** [comment]
@drashna @waffle87 still is there any modifications in this PR? please kindly help to review or merge it, thanks.

**Pull request merged**